events/filters.py

```diff
@@ -0,0 +1,51 @@
+from django.contrib.gis.geos import Point
+from django.contrib.gis.db.models.functions import Distance
+import django_filters
+
+from .models import Event, Category
+
+
+class Filters(django_filters.FilterSet):
+
+    name = django_filters.CharFilter(
+        field_name='name', 
+        lookup_expr='icontains'
+    )
+    category = django_filters.ModelChoiceFilter(
+        queryset=Category.objects.all()
+    )
+    registration_start_date = django_filters.IsoDateTimeFilter(
+        field_name='registration_start_time', 
+        lookup_expr='gte'
+    )
+    registration_end_date = django_filters.IsoDateTimeFilter(
+        field_name='registration_end_time', 
+        lookup_expr='lte'
+    )
+    event_start_date = django_filters.IsoDateTimeFilter(
+        field_name='schedules__start_time', 
+        lookup_expr='gte'
+    )
+    event_end_date = django_filters.IsoDateTimeFilter(
+        field_name='schedules__end_time', 
+        lookup_expr='lte'
+    )
+    location = django_filters.CharFilter(method='filter_by_location')
+
+    class Meta:
+        model = Event
+        fields = [
+            'name', 'category', 'registration_start_date', 
+            'registration_end_date', 'event_start_date', 'event_end_date', 'location'
+        ]
+
+    def filter_by_location(self, queryset, name, value):
+        try:
```

Comment:
try except block should be as small as possible and should only have necessary code which could raise potential error